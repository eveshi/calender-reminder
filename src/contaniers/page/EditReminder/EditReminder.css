*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.EditReminder {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #060505; }
  .EditReminder_header {
    width: 100%;
    height: 15vh;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    display: flex;
    justify-content: space-between; }
    .EditReminder_header a {
      border-radius: 50%;
      background-color: #BD924B;
      height: 5rem;
      width: 5rem;
      position: relative;
      transition: all .2s ease; }
      .EditReminder_header a .Button {
        background-color: transparent; }
      .EditReminder_header a svg {
        position: absolute;
        left: .5rem;
        top: .5rem; }
      .EditReminder_header a:hover {
        transform: scale(0.98); }
  .EditReminder_form {
    height: 85vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center; }
    .EditReminder_form > * {
      margin-bottom: 4rem; }
    .EditReminder_form select {
      font-size: 3.6rem;
      color: #BD924B; }
    .EditReminder_form_input {
      width: 80%;
      display: flex;
      flex-direction: column; }
      .EditReminder_form_input p {
        color: #5F513F;
        margin-top: 3px;
        font-size: 1.8rem;
        text-align: right; }
