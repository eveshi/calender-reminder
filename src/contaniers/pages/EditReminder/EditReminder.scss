@import '../../../stylesheets/main.scss';

.EditReminder {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $color-primary-dark;

  &_header {
    width: 100%;
    height: 15vh;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    display: flex;
    justify-content: space-between;

    @include link_button
  }

  &_form {
    height: 85vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    & > * {
      margin-bottom: 4rem;
    }

    select {
      font-size: 3.6rem;
      color: $color-secondary-dark;
    }

    &_input {
      width: 80%;
      display: flex;
      flex-direction: column;

      p {
        color: $color-secondary-darker;
        margin-top: 3px;
        font-size: 1.8rem;
        text-align: right;
      }
    }
  }
}