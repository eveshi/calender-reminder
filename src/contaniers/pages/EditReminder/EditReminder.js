import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ConnectedDatePicker from '../../../components/DatePicker/DatePicker';
import ConnectedTimePicker from '../../../components/TimePicker/TimePicker';
import Input from '../../../components/Input/Input';
import Button from '../../../components/Button/Button';

import * as actions from '../../../store/actions';

import Close from '../../../assets/icon/Close';
import Complete from '../../../assets/icon/Complete';

import './EditReminder.css';

export const EditReminder = (props) => {
  const {
    yearPicked,
    monthPicked,
    dayPicked,
    preDate,
    preID,
    preTime,
    hourPicked,
    minutePicked,
    datePicked,
    timePicked,
    cleanSession,
    reminder,
    postReminder,
    putReminder,
    changeSessionReminder,
    changeDay,
    changeYearAndMonth,
  } = props;

  const reminderChangeHandler = (event) => {
    const { value } = event.target;
    const newReminder = {
      id: preID,
      time: preTime,
      reminder: value,
    };
    changeSessionReminder(newReminder);
  };

  const submitReminder = () => {
    // const date = (sessionDate === null || dateFns.isPast(sessionDate))
    //   ? utility.formatCurrentDate
    //   : sessionDate;

    // const defaultTime = dateFns.isPast(sessionDate)
    //   ? utility.timeIDGenerator(dateFns.getHours(new Date()), utility.minuteFiveGenerator())
    //   : 0;

    // const time = sessionTime === null
    //   ? defaultTime
    //   : sessionTime;

    if (preID === 0) {
      postReminder(datePicked, timePicked, reminder);
    } else {
      putReminder(preDate, datePicked, preID, timePicked, reminder);
    }
    changeDay(dayPicked);
    changeYearAndMonth(yearPicked, monthPicked);
    cleanSession();
  };

  return (
    <div className="EditReminder">
      <div className="EditReminder_header">
        <Link to="/">
          <Button onClick={() => cleanSession()}>
            <Close />
          </Button>
        </Link>
        <Link to="/">
          <Button onClick={() => submitReminder()}>
            <Complete />
          </Button>
        </Link>
      </div>
      <div className="EditReminder_form">
        <ConnectedDatePicker
          dayPickerDisabled={false}
          datePickerDisabled={false}
        />
        <ConnectedTimePicker
          hourInit={hourPicked}
          minuteInit={minutePicked}
          datePicked={datePicked}
        />
        <div className="EditReminder_form_input">
          <Input
            maxLength="50"
            placeholder="Dinner with Danny..."
            value={reminder}
            onChange={event => reminderChangeHandler(event)}
          />
          <p>
            {reminder.length}
            /50
          </p>
        </div>
      </div>
    </div>
  );
};

EditReminder.propTypes = {
  yearPicked: PropTypes.number.isRequired,
  monthPicked: PropTypes.number.isRequired,
  dayPicked: PropTypes.number.isRequired,
  preDate: PropTypes.string.isRequired,
  preID: PropTypes.number.isRequired,
  preTime: PropTypes.number.isRequired,
  hourPicked: PropTypes.string.isRequired,
  minutePicked: PropTypes.string.isRequired,
  datePicked: PropTypes.string.isRequired,
  timePicked: PropTypes.number.isRequired,
  reminder: PropTypes.string.isRequired,
  cleanSession: PropTypes.func.isRequired,
  postReminder: PropTypes.func.isRequired,
  putReminder: PropTypes.func.isRequired,
  changeSessionReminder: PropTypes.func.isRequired,
  changeDay: PropTypes.func.isRequired,
  changeYearAndMonth: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  yearPicked: state.sessionState.year,
  monthPicked: state.sessionState.month.index,
  dayPicked: state.sessionState.day,
  preTime: state.sessionState.reminder.time,
  preID: state.sessionState.reminder.id,
  reminder: state.sessionState.reminder.reminder,
  preDate: state.reminderState.date.date,
  hourPicked: state.sessionState.hour,
  minutePicked: state.sessionState.minute,
  datePicked: state.sessionState.date,
  timePicked: state.sessionState.time,
});

const mapActionToProps = dispatch => ({
  postReminder: (date, time, reminder) => dispatch(actions.postReminder(date, time, reminder)),
  putReminder: (preDate, nextDate, id, time, reminder) => dispatch(
    actions.putReminder(preDate, nextDate, id, time, reminder),
  ),
  cleanSession: () => dispatch(actions.cleanSession()),
  changeSessionReminder: reminder => dispatch(actions.changeSessionReminder(reminder)),
  changeDay: day => dispatch(actions.changeDay(day)),
  changeYearAndMonth: (year, month) => dispatch(actions.changeYearAndMonth(year, month)),
});

export default connect(mapStateToProps, mapActionToProps)(EditReminder);
