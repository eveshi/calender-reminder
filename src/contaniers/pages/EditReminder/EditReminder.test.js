import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import { EditReminder } from './EditReminder';

configure({ adapter: new Adapter() });

const props = {
  yearPicked: 2019,
  monthPicked: 8,
  dayPicked: 3,
  preDate: '20200312',
  preID: 1330515905123,
  preTime: 1604,
  hourPicked: '00',
  minutePicked: '00',
  datePicked: '20190803',
  timePicked: 1708,
  reminder: 'Dinner',
  cleanSession: jest.fn(),
  postReminder: jest.fn(),
  putReminder: jest.fn(),
  changeSessionReminder: jest.fn(),
  changeDay: jest.fn(),
  changeYearAndMonth: jest.fn(),
};

jest.mock('../../../components/DatePicker/DatePicker.js', () => () => (
  <div id="DatePicker">
    DatePicker
  </div>));

jest.mock('../../../components/TimePicker/TimePicker.js', () => () => (
  <div id="TimePicker">
    TimePicker
  </div>));

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <EditReminder {...props} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
