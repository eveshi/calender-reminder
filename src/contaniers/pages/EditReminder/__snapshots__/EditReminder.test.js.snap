// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div
  className="EditReminder"
>
  <div
    className="EditReminder_header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="Button"
        disabled={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="200"
          viewBox="0 0 1024 1024"
          width="200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="add_reminder"
          >
            DELETE
          </title>
          <path
            d="M809.984 274.005333l-237.994667 237.994667 237.994667 237.994667-59.989333 59.989333-237.994667-237.994667-237.994667 237.994667-59.989333-59.989333 237.994667-237.994667-237.994667-237.994667 59.989333-59.989333 237.994667 237.994667 237.994667-237.994667z"
          />
        </svg>
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="Button"
        disabled={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="200"
          viewBox="0 0 1024 1024"
          width="200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="add_reminder"
          >
            DELETE
          </title>
          <path
            d="M384 690.005333l452.010667-452.010667 59.989333 59.989333-512 512-237.994667-237.994667 59.989333-59.989333z"
          />
        </svg>
      </button>
    </a>
  </div>
  <div
    className="EditReminder_form"
  >
    <div
      id="DatePicker"
    >
      DatePicker
    </div>
    <div
      id="TimePicker"
    >
      TimePicker
    </div>
    <div
      className="EditReminder_form_input"
    >
      <input
        className="Input"
        maxLength="50"
        onChange={[Function]}
        onKeyPress={null}
        placeholder="Dinner with Danny..."
        type="text"
        value="Dinner"
      />
      <p>
        6
        /50
      </p>
    </div>
  </div>
</div>
`;
